driver-core/src/test/unit/com/mongodb/AuthConnectionStringTest.java

```diff
@@ -133,27 +140,32 @@ private void assertMechanism(final String key, final String actual) {
 
     private void assertMechanismProperties(final MongoCredential credential) {
         BsonValue expected = getExpectedValue("credential.mechanism_properties");
-
-        if (!expected.isNull()) {
-            BsonDocument document = expected.asDocument();
-            for (String key : document.keySet()) {
-                if (document.get(key).isString()) {
-                    String expectedValue = document.getString(key).getValue();
-
-                    // If the mechanism is "GSSAPI", the default SERVICE_NAME, which is stated as "mongodb" in the specification,
-                    // is set to null in the driver.
-                    if (credential.getMechanism().equals("GSSAPI") && key.equals("SERVICE_NAME") && expectedValue.equals("mongodb")) {
-                        assertNull(credential.getMechanismProperty(key, null));
-                    } else {
-                        assertEquals(expectedValue, credential.getMechanismProperty(key, null));
-                    }
+        if (expected.isNull()) {
```

Comment:
just a question, do you validate this option https://github.com/mongodb/specifications/blob/master/source/auth/tests/legacy/connection-string.json#L487?

Reply:
I believe I do read that field, but it is not in this PR since it is just OIDC-independent refactoring that I extracted from my current work.